### PR TITLE
Use deployment instead of statefulsets for sidecar controllers 

### PR DESCRIPTION
**Pull request by nitishkumar4**
## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

- Sidecar controller pods were deployed using Kubernetes Statefulset resource.

## What is the new behavior?

- Sidecar controllers are now deployed using Kubernetes Deployment resource.

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by nitishkumar4** [comment]
Private image: quay.io/nitish3356/csi:operator281705

File: operator/api/v1/csiscaleoperator_types.go [hunk 1]
```diff
@@ -387,11 +387,12 @@ func init() {
 type CSIReason string
 
 const (
-	CSIConfigured       CSIReason = "Successful"
+	CSIConfigured       CSIReason = "successful"
```
Comment (deeghuge) [inline operator/api/v1/csiscaleoperator_types.go:390]: Hi @nitishkumar4 , Why do we need to change these errors now ?

Comment (nitishkumar4) [reply operator/api/v1/csiscaleoperator_types.go:390]: These strings are updated in Status section, do you think other teams automation would depend on the string case used here? If so, then we should revert back to as it was for now.

**Review by madhuthorat — APPROVED** [review]
LGTM

**Review by amdabhad — APPROVED** [review]
LGTM

**Pull request merged**